lib/internal/modules/esm/resolve.js

```diff
@@ -1068,15 +1068,11 @@ function defaultResolve(specifier, context = {}) {
     // Avoid accessing the `protocol` property due to the lazy getters.
     protocol = parsed.protocol;
 
-    if (protocol === 'data:' &&
-      parsedParentURL.protocol !== 'file:' &&
-      experimentalNetworkImports) {
-      throw new ERR_NETWORK_IMPORT_DISALLOWED(
-        specifier,
-        parsedParentURL,
-        'import data: from a non file: is not allowed',
-      );
-    }
+   if (protocol === 'data:' && parentProtocol !== 'file:' && parentProtocol !== 'data:') {
+  throw new ERR_NETWORK_IMPORT_DISALLOWED(
+    'import of \'data:\' is not supported: import data: from a non file: is not allowed'
+  );
```

Comment:
```suggestion
   if (protocol === 'data:' && parentProtocol !== 'file:' && parentProtocol !== 'data:') {
     throw new ERR_NETWORK_IMPORT_DISALLOWED(
       'import of \'data:\' is not supported: import data: from a non file: is not allowed'
  );
```

Reply:
While it's not related to this PR, we should refine the check and the error message to disallow importing from network modules, there are no reasons to forbid importing from custom protocols.
```suggestion
   if (protocol === 'data:' && (parentProtocol === 'http:' && parentProtocol === 'https:')) {
     throw new ERR_NETWORK_IMPORT_DISALLOWED(
       'import of \'data:\' is not supported: import data: from a remote location is not allowed'
     );
```